app/src/features/datasets/components/security/ApplySecurityDialog.tsx

```diff
@@ -0,0 +1,176 @@
+          validateOnChange={false}
+          onSubmit={async (values: { securityReasons: ISecurityReason[] }) => {
+            await handleSubmit(values.securityReasons);
+            showSnackBar({
+              snackbarMessage: (
+                <>
+                  <Typography variant="body2" component="div">
+                    You successfully applied security reasons to the file{selectedArtifacts.length > 1 && 's'}.
+                  </Typography>
+                </>
+              ),
+              open: true
+            });
+            onClose();
+          }}>
+          {(formikProps) => (
+            <>
+              <YesNoDialog
+                open={yesNoDialogOpen}
+                onClose={() => setYesNoDialogOpen(false)}
+                onYes={async () => {
+                  await formikProps.submitForm();
+                  showSnackBar({
+                    snackbarMessage: (
+                      <>
+                        <Typography variant="body2" component="div">
+                          You successfully unsecured the file{selectedArtifacts.length > 1 && 's'}.
+                        </Typography>
+                      </>
+                    ),
+                    open: true
+                  });
+                  setYesNoDialogOpen(false);
+                }}
+                onNo={() => setYesNoDialogOpen(false)}
+                dialogTitle=""
+                dialogText=""
+                dialogContent={
+                  <Box sx={{ mb: 2, display: 'flex', flexDirection: 'column', alignContent: 'center' }}>
+                    <Box sx={{ display: 'flex', flexDirection: 'row', justifyContent: 'center' }}>
+                      <Icon path={mdiInformationOutline} size={5} color={'#a12622'} />
+                    </Box>
+                    <DialogTitle id="component-dialog-title" align={'center'}>
+                      Warning
+                    </DialogTitle>
+                    <DialogContentText id="alert-dialog-description" align={'center'} sx={{ py: 2 }}>
+                      You are going to make this document available to the public. This document can be downloaded.
+                      Also, if there are any security reasons, they will be removed.
+                    </DialogContentText>
+                    <DialogContentText id="alert-dialog-description" align={'center'}>
+                      <strong>Are you sure you would like to proceed?</strong>
+                    </DialogContentText>
+                  </Box>
+                }
+              />
+
+              <DialogTitle id="component-dialog-title">Apply Security Reasons</DialogTitle>
+              <DialogContent sx={{ display: 'flex', flexDirection: 'column', overflow: 'hidden' }}>
+                <DialogContentText id="alert-dialog-description">
+                  Search for the security reasons and apply them to the selected document
+                  {selectedArtifacts.length > 1 ? 's' : ''}.
```

Comment:
This check will append an "s" when zero documents are selected. Instead the check should be `selectedArtifacts.length !== 1 ? 's' : ''`

Reply:
Can do that, the dialog is in accessible without selecting at least one artifact.